tools/src/commands/CherryPick.ts

```diff
@@ -0,0 +1,196 @@
+        commitsOnDestinationBranch.some(
+          (destCommit) =>
+            srcCommit.authorDate === destCommit.authorDate &&
+            srcCommit.authorName === destCommit.authorName &&
+            srcCommit.title === destCommit.title
+        )
+      ) {
+        return false;
+      }
+
+      // Filter out any commits earlier than the startDate (if we have one) but also add them to
+      // another array so we can log them.
+      if (startDate && new Date(srcCommit.authorDate).getTime() < startDate.getTime()) {
+        commitsBeforeStartDate.push(srcCommit);
+        return false;
+      }
+
+      return true;
+    });
+
+  if (commitsBeforeStartDate.length !== 0) {
+    logger.log(chalk.bold(chalk.red('Ignoring the following commits from before the start date:')));
+    logger.log(
+      commitsBeforeStartDate
+        .map(
+          (commit) =>
+            ` ❌ ${chalk.red(commit.hash.slice(0, 10))} ${commit.authorDate} ${chalk.magenta(
+              commit.authorName
+            )} ${commit.title}`
+        )
+        .join('\n')
+    );
+    logger.log(''); // new line
+  }
+
+  if (candidateCommits.length === 0) {
+    logger.success('There is nothing to cherry-pick.');
+    return;
+  }
+
+  const { commitsToCherryPick } = await inquirer.prompt<{ commitsToCherryPick: string[] }>([
+    {
+      type: 'checkbox',
+      name: 'commitsToCherryPick',
+      message: `Choose which commits to cherry-pick from ${chalk.blue(source)} to ${chalk.blue(
+        destination
+      )}\n  ${chalk.green('●')} selected  ○ unselected\n`,
+      choices: candidateCommits.map((commit) => ({
+        value: commit.hash,
+        short: commit.hash,
+        name: `${chalk.yellow(commit.hash.slice(0, 10))} ${commit.authorDate} ${chalk.magenta(
+          commit.authorName
+        )} ${commit.title}`,
+      })),
+      default: candidateCommits.map((commit) => commit.hash),
+      pageSize: Math.min(candidateCommits.length, (process.stdout.rows || 100) - 4),
+    },
+  ]);
+
+  logger.info(''); // new line
+
+  if (destination !== (await Git.getCurrentBranchNameAsync())) {
+    if (options.dry) {
+      logger.log(chalk.bold(chalk.yellow(`git checkout ${destination}`)));
+    } else {
+      logger.info(`Checking out ${chalk.bold(chalk.blue(destination))} branch...`);
+      await Git.checkoutAsync(destination);
+    }
+  }
+
+  // ensure we preserve the correct order of commits
+  const commitsToCherryPickOrdered = candidateCommits.filter((commit) =>
+    commitsToCherryPick.includes(commit.hash)
+  );
+  const gitArgs = ['cherry-pick', ...commitsToCherryPickOrdered.map((commit) => commit.hash)];
+  if (options.dry) {
+    logger.log(chalk.bold(chalk.yellow(`git ${gitArgs.join(' ')}`)));
+  } else {
+    logger.info(`Running ${chalk.yellow(`git ${gitArgs.join(' ')}`)}`);
+    // pipe output to current process stdio to emulate user running this command directly
+    await Git.runAsync(gitArgs, { stdio: 'inherit' }).catch(() =>
```

Comment:
Does it make sense to instead add the `cherryPickAsync` function to the `Git` object?

Also, you're mixing promises' `async/await` style with `then/catch` — maybe better to do something like 👇?
```
try {
  await Git.runAsync(...);
} catch {
  logger.error(...);
}
```

Reply:
Sure, may as well go ahead and add it -- I didn't at first because of the extra SpawnOptions arg I wanted to pass in, but I suppose it may be useful for other tools in the future that want to call `cherryPickAsync`.